core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -152,43 +140,38 @@ class LoadBalancerService(
      * Creates an activation entry and insert into various maps.
      */
     private def setupActivation(action: ExecutableWhiskAction, activationId: ActivationId, namespaceId: UUID, invokerName: String, transid: TransactionId): ActivationEntry = {
-        // either create a new promise or reuse a previous one for this activation if it exists
+
         val timeout = action.limits.timeout.duration + activeAckTimeoutGrace
```

Comment:
The invariant now is that the activation is setup exactly once (otherwise there will be duplicates); is this enforced somewhere? Maybe a require statement is warranted.